src/domain/settings/currencies.js

```diff
@@ -1,168 +1,116 @@
+import useMedusa from "../../hooks/use-medusa"
 import { currencies } from "../../utils/currencies"
-import { Label } from "@rebass/forms"
 import { getErrorMessage } from "../../utils/error-messages"
 
-const StyledMultiSelect = styled(MultiSelect)`
-  ${Typography.Base}
-
-  color: black;
-  background-color: white;
-
-  width: 150px;
-
-  line-height: 1.22;
+const CurrencySettings = () => {
+  const [storeCurrencies, setStoreCurrencies] = useState([])
+  const [allCurrencies, setAllCurrencies] = useState([])
+  const [selectedCurrency, setSelectedCurrency] = useState({
+    value: "",
+    label: "",
+  })
 
-  border: none;
-  outline: 0;
+  const { store, isLoading, update, toaster, refresh } = useMedusa("store")
 
-  transition: all 0.2s ease;
+  const setCurrenciesOnLoad = () => {
+    const defaultCurr = {
+      label: store.default_currency_code.toUpperCase(),
+      value: store.default_currency_code.toUpperCase(),
+    }
 
-  border-radius: 3px;
-  box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px 0px, rgba(0, 0, 0, 0) 0px 0px 0px 0px,
-    rgba(0, 0, 0, 0) 0px 0px 0px 0px, rgba(60, 66, 87, 0.16) 0px 0px 0px 1px,
-    rgba(0, 0, 0, 0) 0px 0px 0px 0px, rgba(0, 0, 0, 0) 0px 0px 0px 0px,
-    rgba(0, 0, 0, 0) 0px 0px 0px 0px;
+    const storeCurrs = store.currencies.map(c => ({
+      value: c.code.toUpperCase(),
+      label: c.code.toUpperCase(),
+    }))
 
-  &:focus: {
-    box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px 0px,
-      rgba(206, 208, 190, 0.36) 0px 0px 0px 4px,
-      rgba(0, 0, 0, 0) 0px 0px 0px 0px, rgba(60, 66, 87, 0.16) 0px 0px 0px 1px,
-      rgba(0, 0, 0, 0) 0px 0px 0px 0px, rgba(0, 0, 0, 0) 0px 0px 0px 0px,
-      rgba(0, 0, 0, 0) 0px 0px 0px 0px;
-  }
-  &::placeholder: {
-    color: #a3acb9;
-  }
+    const allCurrs = Object.keys(currencies).map(currency => ({
+      label: currency,
+      value: currency,
+    }))
 
-  .go3433208811 {
-    border: none;
-    border-radius: 3px;
+    setSelectedCurrency(defaultCurr)
+    setStoreCurrencies(storeCurrs)
+    setAllCurrencies(allCurrs)
   }
-`
-
-const AccountDetails = () => {
-  const [selectedCurrencies, setCurrencies] = useState([])
-  const { register, setValue, handleSubmit } = useForm()
-  const { store, isLoading, update, toaster } = useMedusa("store")
 
   useEffect(() => {
-    if (isLoading || !store) return
-    setValue("default_currency_code", store.default_currency_code.toUpperCase())
-    setCurrencies(
-      store.currencies
-        ? store.currencies.map(c => ({
-            value: c.code.toUpperCase(),
-            label: c.code.toUpperCase(),
-          }))
-        : []
-    )
-  }, [store, isLoading])
-
-  const options = Object.keys(currencies).map(k => {
-    return {
-      value: k,
-      label: k,
+    if (isLoading || !store) {
+      return
     }
-  })
+
+    setCurrenciesOnLoad()
+  }, [store, isLoading])
 
   const handleChange = currencies => {
-    setCurrencies(currencies)
+    setStoreCurrencies(currencies)
```

Comment:
Dont know what the expected behaviour is here, but right now you can remove the default store currency from the store-currencies and save the resulting currency configuration:

![image](https://user-images.githubusercontent.com/88927411/149282682-8c8bfb4a-7dd5-4379-97e9-c2bc7b019fb7.png)

We should probably enforce the behvaiour both here and in the core (i have created a ticket)

Reply:
Should be fixed as part of core [PR](https://github.com/medusajs/medusa/pull/984) 👍